api/src/pcapi/core/educational/api/dms.py

```diff
@@ -0,0 +1,165 @@
+from datetime import datetime
+from datetime import timedelta
+import logging
+import typing
+
+from pcapi import settings
+from pcapi.connectors.dms import api as dms_api
+from pcapi.connectors.dms import models as dms_models
+from pcapi.core.educational import models as educational_models
+from pcapi.core.offerers import models as offerers_models
+from pcapi.models import db
+
+
+logger = logging.getLogger(__name__)
+
+
+def update_dms_status(procedure_id: int = 0) -> None:
+    """import dms applications for eac status"""
+    if procedure_id:
+        procedures = [procedure_id]
+    else:
+        procedures = [
+            settings.DMS_EAC_PROCEDURE_INDEPENDANTS_CANDIDATE_ID,
+            settings.DMS_EAC_PROCEDURE_STRUCTURE_CANDIDATE_ID,
+            settings.DMS_EAC_PROCEDURE_MENJS_CANDIDATE_ID,
+        ]
+
+    for procedure in procedures:
+        previous_import = _get_previous_import(procedure)
+        if previous_import and previous_import.isProcessing:
+            logger.info(
+                "[DMS] Procedure %s is already being processed.",
+                procedure,
+                extra={
+                    "procedure_number": procedure,
+                },
+            )
+            continue
+        since = previous_import.latestImportDatetime if previous_import else None
+
+        _update_dms_status_for_procedure(
+            procedure_number=procedure,
+            since=since,
+        )
+
+
+def _update_dms_status_for_procedure(procedure_number: int, since: datetime | None) -> None:
+    logger.info(
+        "[DMS] Starting to process procedure %s.",
+        procedure_number,
+        extra={
+            "procedure_number": procedure_number,
+        },
+    )
+
+    current_import = dms_models.LatestDmsImport(
+        procedureId=procedure_number,
+        latestImportDatetime=datetime.utcnow(),
+        isProcessing=True,
+        processedApplications=[],
+    )
+    db.session.add(current_import)
+    db.session.commit()
+
+    dms_client = dms_api.DMSGraphQLClient()
+    applications_id = []
+
+    for node in dms_client.get_eac_nodes_siret_states(procedure_id=procedure_number, since=since):
+        try:
+            _get_or_create_application(procedure_number=procedure_number, node=node)
+        except Exception as exp:  # pylint: disable=broad-exception-caught
+            logger.error(
```

Comment:
(détail) Autant utiliser `logger.exception() pour avoir la stack trace complète dans Sentry, ça peut toujours être utile.

```suggestion
            logger.exception(
```

Reply:
Je croyais que c'était error qui le faisait. Merci pour la correction